src/libAtomVM/refc_binary.c

```diff
@@ -76,7 +76,7 @@ term refc_binary_create_binary_info(Context *ctx)
     if (len == 0) {
         return term_nil();
     }
-    if (memory_ensure_free(ctx, len * TUPLE_SIZE(2)) != MEMORY_GC_OK) {
+    if (memory_ensure_free(ctx, len * (2 + TUPLE_SIZE(2))) != MEMORY_GC_OK) {
```

Comment:
Not for this PR, but I wonder if we should add CONS_SIZE in the future for readability.